interfaces/builtin/README.md

```diff
@@ -0,0 +1,308 @@
+in a web-streaming application).
+
+The default value of the `allow-*` keys when not otherwise specified is `true`
+and the default value of `deny-*` keys when not otherwise specified is `false`.
+Matching `deny-*` constraints override/take precedence over `allow-*`
+constraints.
+
+
+### allow-installation
+
+The `allow-installation` key is the first key that is evaluated when the snap
+is being installed. If this evaluates to false, the snap cannot be installed if
+an interface plug or slot for which `allow-installation` evaluated to `false`
+exists in the snap. An example would be the `snapd-control` interface, which
+has in the base-declaration the static `allow-installation: false` rule for
+plugs:
+
+    snapd-control:
+      allow-installation: false
+      deny-auto-connection: true
+
+If a snap does not plug `snapd-control` then this rule does not apply, but if
+the snap does declare a `snapd-control` plug and there are no assertions in the
+store for this snap about allowing `snapd-control`, then snap installation will
+fail.
+
+Snap interfaces which have `allow-installation` set to `false` for their plugs
+in the base-declaration are said to be “**super-privileged**”, meaning they
+cannot be used at all without a snap-declaration assertion.
+
+
+### allow-connection
+
+The `allow-connection` key is the next key to be evaluated after the snap is
+allowed to be installed. This key controls whether a connection is permitted at
+all and usually is used to ensure that only “compatible” plugs and slots are
+connected to each other. A great example is the content interface, where the
+following (abbreviated) rule from the base-declaration is used to ensure that a
+candidate plug and slot content interface have matching `content` attribute
+values:
+
+    allow-connection:
+      plug-attributes:
+        content: $SLOT(content)
+
+This can be read as `allow-connection` evaluating to `true` only when the plug
+has an attribute `content` with the same value as the attribute `content` in
+the slot. That is to say, these plug and slots are compatible because `content`
+does match for the plug and slot:
+
+    # in the snap providing the content:
+    slots:
+      foo-content:
+        interface: content
+        content: specific-files
+
+    # in the snap consuming the content:
+    plugs:
+      foo-content:
+        interface: content
+        content: specific-files
+
+While the following plug and slots are not compatible:
+
+    slots:
+      foo-content:
+        interface: content
+        content: other-files
+
+    plugs:
+      foo-content:
+        interface: content
+        content: specific-files
+
+
+### allow-auto-connection
+
+The allow-auto-connection key is the final key considered when snapd is
+evaluating automatic connection of interface plugs and slots. If this key
+evaluates to `true`, then this plug/slot combination is considered a valid
+candidate for automatic connection.
```

Comment:
...automatic connection will happen normally only if there is one candidate pairing with a slot for a given plug.